client/galaxy/scripts/mvc/tool/tool-form-base.js

```diff
@@ -151,19 +94,9 @@ define( [ 'utils/utils', 'utils/deferred', 'mvc/ui/ui-misc', 'mvc/form/form-view
                             icon    : 'fa-cube',
                             onclick : function() {
                                 // here we update the tool version (some tools encode the version also in the id)
-                                var id = options.id.replace( options.version, this.version );
-                                var version = this.version;
-                                // queue model request
-                                self.deferred.reset();
-                                self.deferred.execute( function( process ) {
-                                    if ( options.hasOwnProperty( "workflow" ) ) {
-                                        // this is needed for notifying the workflow editor form
-                                        options.version = version;
-                                        self.model.get( 'postchange' )( process, self );
-                                    } else {
-                                        self._buildModel( process, { id : id, version : version } );
-                                    };
-                                });
+                                self.model.set( 'id', options.id.replace( options.version, this.version ) );
```

Comment:
Yeah, I am not a fan of having to replace the id in the string either. This PR does not change that tho. We introduced it because the tool version is encoded in the tool id / url of tool shed tools.